packages/core/src/stateUtils.ts

```diff
@@ -640,64 +598,58 @@ export function resolveTarget<
 function resolveHistoryTarget<
   TContext extends MachineContext,
   TEvent extends EventObject
->(
-  stateNode: StateNode<TContext, TEvent> & { type: 'history' }
-): Array<StateNode<TContext, TEvent>> {
+>(stateNode: AnyStateNode & { type: 'history' }): Array<AnyStateNode> {
   const normalizedTarget = normalizeTarget<TContext, TEvent>(stateNode.target);
   if (!normalizedTarget) {
     return stateNode.parent!.initial.target;
   }
   return normalizedTarget.map((t) =>
-    typeof t === 'string'
-      ? getStateNodeByPath<TContext, TEvent>(stateNode, t)
-      : t
+    typeof t === 'string' ? getStateNodeByPath(stateNode, t) : t
   );
 }
 
-function isHistoryNode<
-  TContext extends MachineContext,
-  TEvent extends EventObject
->(
-  stateNode: StateNode<TContext, TEvent>
-): stateNode is StateNode<TContext, TEvent> & { type: 'history' } {
+function isHistoryNode(
+  stateNode: AnyStateNode
+): stateNode is AnyStateNode & { type: 'history' } {
   return stateNode.type === 'history';
 }
 
-export function getInitialStateNodes<
-  TContext extends MachineContext,
-  TEvent extends EventObject
->(stateNode: StateNode<TContext, TEvent>): Array<StateNode<TContext, TEvent>> {
-  const transitions = [
-    {
-      target: [stateNode],
-      source: stateNode,
-      actions: [],
-      eventType: 'init',
-      toJSON: null as any // TODO: fix
+export function getInitialStateNodes(
```

Comment:
You have stopped using `computeEntrySet` here and thus you no longer "dereference" default targets of history states.

This is a regression for the test [here](https://github.com/statelyai/xstate/pull/3723/files#diff-2794cd100b6e979138d1948c83f08af5544ce19da63aa17945149482bdc1e22cR156). Note that this test also depends on the fix from the same PR ([this one](https://github.com/statelyai/xstate/pull/3723/files#diff-6f3a03f5c8d5024e53467d74c187685ab0acd0e1efb2549fbf2842d9616ad8dbR652))

Reply:
Can we bring these tests into this PR?

Reply:
This PR has been merged to `next` - so you just need to merge `next` to this mega PR here